app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,262 @@
+  },
+  fhirIconCell: {
+    boxShadow: `inset -1px 0 0 ${theme.palette.divider}`,
+    position: "sticky",
+    left: 0,
+    padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
+    width: 25,
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  cells: {
+    whiteSpace: "nowrap",
+    boxShadow: `inset -1px 0 0 ${theme.palette.divider}`,
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  texts: {
+    textAlign: "center",
+  },
+  cellsTitle: {
+    background: theme.palette.background.paper,
+  },
+  rowBorder: {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  preview: {
+    padding: 0,
+    width: "100%",
+    borderRadius: theme.shape.borderRadius,
+    border: `1px solid ${theme.palette.divider}`,
+    "& > div": {
+      padding: theme.spacing(3),
+      borderRadius: theme.shape.borderRadius,
+    },
+  },
+  button: {
+    alignSelf: "flex-start",
+    marginBottom: theme.spacing(2),
+    textTransform: "none",
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+    },
+  },
+}));
+
+type Exploration = { fields: string[]; rows: string[][] };
+
+const Preview = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const history = useHistory();
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const { mappingId } = useParams<{
+    mappingId?: string;
+  }>();
+
+  const [alert, setAlert] = useState<string | undefined>(undefined);
+  const handleAlertClose = () => setAlert(undefined);
+
+  const [preview, setPreview] = useState<Record<string, unknown> | undefined>(
+    undefined
+  );
+
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: mappingId === undefined }
+  );
+
+  const { data: owner } = useApiOwnersRetrieveQuery(
+    {
+      id: mapping?.primary_key_owner ?? "",
+    },
+    { skip: mapping === undefined }
+  );
+
+  const {
+    data: exploration,
+    isLoading: isExplorationLoading,
+  } = usePagaiExploreRetrieveQuery(
+    {
+      resourceId: mapping?.id ?? "",
+      owner: owner?.name ?? "",
+      table: mapping?.primary_key_table ?? "",
+    },
+    { skip: mapping === undefined || owner === undefined }
+  );
+
+  const [apiPreviewCreate] = useApiPreviewCreateMutation();
+
+  const handleFhirIconClick = (index: number) => async () => {
+    const primaryKey = mapping?.primary_key_table;
+    if (exploration && primaryKey && mapping) {
+      const indexPrimaryKey = (exploration as Exploration).fields.indexOf(
+        primaryKey
+      );
+      const primaryKeyValue = (exploration as Exploration).rows[index]?.[
+        indexPrimaryKey
+      ];
+      try {
+        const previewResult = await apiPreviewCreate({
+          previewRequest: {
+            resource_id: mapping?.id,
+            primary_key_values: [primaryKeyValue ?? ""],
+          },
+        });
```

Comment:
No need to call `.unwrap` with this mutation hook ? 🤔

Reply:
oops good catch